controllers/chat/sendSignedMessage.js

```diff
@@ -131,8 +132,43 @@ const sendSignedMesage = async (req, res) => {
       });
     }
 
-    const chat = await channel.sendMessage(baseMessage);
+    const chat = await channel.sendMessage(baseMessage, {skip_push: true});
+    let notificationPayload = {
+      title: `New Message from ${senderInfo?.user?.username}`,
+      body: `${message.substring(0, 100)}`
+    };
+    let dataPayload = {
+      channel_id: channelId,
+      user_id: req.userId,
+      message_id: chat.message.id,
+      message,
+      message_schema: currentMessageType,
+      type: 'message.new',
+      status: 'sent',
+      is_big_message: 'false',
+      attachment: JSON.stringify(attachments) ?? '',
+      created_at: chat.message.created_at,
+      is_annoymous: 'false'
+    };
+    const dataSizeInBytes = Buffer.byteLength(JSON.stringify(dataPayload), 'utf8');
+    const dataSizeInKilobytes = dataSizeInBytes / 1024;
+    if (dataSizeInKilobytes > 4) {
+      dataPayload.is_big_message = 'true';
+      dataPayload.attachment = '';
+    }
 
+    channelMember.forEach((member) => {
+      let payload = {
+        notification: notificationPayload,
+        // priority: 'high',
+        // content_available: true,
+        data: {
+          ...dataPayload,
+          is_annoymous: member.is_anonymous.toString()
+        }
+      };
+      BetterSocialCore.fcmToken.sendChatNotification(member.user_id, payload);
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Here, a notification is sent to each member of the channel. However, there's no error handling for when `sendChatNotification` fails. If sending a notification fails for one member, it could prevent notifications from being sent to subsequent members. Consider adding error handling to continue sending notifications to the remaining members even if one fails.


<!-- This is an auto-generated comment by OSS CodeRabbit -->